packages/cli/src/cmds/preview/preview.ts

```diff
@@ -282,41 +288,55 @@ export default class Preview extends YargsCommand {
       );
     }
 
-    // check node
+    const start = performance.now();
     const depsManager = new DepsManager(cliEnvCheckerLogger, cliEnvCheckerTelemetry);
-    const nodeRes = await this.checkNode(includeBackend, includeFuncHostedBot, depsManager);
-    if (nodeRes.isErr()) {
-      return err(nodeRes.error);
-    }
+    try {
+      // check node
+      const nodeRes = await this.checkNode(includeBackend, includeFuncHostedBot, depsManager);
+      if (nodeRes.isErr()) {
+        return err(nodeRes.error);
+      }
 
-    // check account
-    const accountRes = await this.checkM365Account();
-    if (accountRes.isErr()) {
-      return err(accountRes.error);
-    }
+      // check account
+      const accountRes = await this.checkM365Account();
+      if (accountRes.isErr()) {
+        return err(accountRes.error);
+      }
 
-    // check cert
-    const certRes = await this.resolveLocalCertificate(localEnvManager);
-    if (certRes.isErr()) {
-      return err(certRes.error);
-    }
+      // check cert
+      const certRes = await this.resolveLocalCertificate(localEnvManager);
+      if (certRes.isErr()) {
+        return err(certRes.error);
+      }
 
-    // check deps
-    const envCheckerResult = await this.handleDependences(
-      includeBackend,
-      includeBot,
-      includeFuncHostedBot,
-      depsManager
-    );
-    if (envCheckerResult.isErr()) {
-      return err(envCheckerResult.error);
-    }
+      // check deps
+      const envCheckerResult = await this.handleDependences(
+        includeBackend,
+        includeBot,
+        includeFuncHostedBot,
+        depsManager
+      );
+      if (envCheckerResult.isErr()) {
+        return err(envCheckerResult.error);
+      }
 
-    // clear background tasks
-    this.backgroundTasks = [];
-    // init service log writer
-    this.serviceLogWriter = new ServiceLogWriter();
-    await this.serviceLogWriter.init();
+      // clear background tasks
+      this.backgroundTasks = [];
+      // init service log writer
+      this.serviceLogWriter = new ServiceLogWriter();
+      await this.serviceLogWriter.init();
```

Comment:
Move this block outside, just before "start ngrok".

Reply:
Added.